src/lsm/compaction.zig

```diff
@@ -1895,6 +1905,11 @@ pub fn CompactionType(
             // TODO: Can we assert target_values_merge_count > 0 here?
             assert(bar.target_values_merge_count == bar.target_values_write_count);
 
+            // Assert we've finished within the number of beats we were allocated.
+            // TODO(metric): Track the delta between target and actual.
+            if (!bar.move_table)
+                assert(bar.beats_finished <= bar.beats_max.?);
```

Comment:
My reading is that our style consensus is mildly in favor of using braces here ^^